__test__/App.spec.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { BrowserRouter, Switch } from 'react-router-dom';
+import thunk from 'redux-thunk';
+import configureStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+import LoginPage from '@pages/loginPage';
+import HomePage from '@pages/homePage';
+import NotFoundPage from '@pages/notFoundPage';
+import Menu from '../src/components/userMenu';
+import Footer from '../src/components/Footer/index';
+import Auth from '../src/components/auth';
+
+import App from '../src/routes/AppRouter';
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+const store = mockStore({
+  authReducer: {
+    isAuthenticated: false,
+    user: {},
+    status: 'rest'
+  }
+});
+
+const auth = (state = {}, action) => {
+  switch (action.type) {
+    case 'LOGIN_SUCCESS':
+      return {
+        isLoginPending: false,
+        isAuthenticated: !action.user,
+        user: action.user
+      };
+    default:
+      return state;
+  }
+};
+
+describe('Application test', () => {
+  it('should render index page', () => {
+    const comp = (
+      <Provider store={store}>
+        <BrowserRouter>
+          <Switch>
+            <App dispatch={jest.fn()} />
+          </Switch>
+        </BrowserRouter>
+      </Provider>
+    );
+    const wrapper = mount(comp);
+
+    expect(wrapper.find('Home')).toBeTruthy();
+  });
+  it('should not crash app', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <App />
+      </Provider>
+    );
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+  it('should work fine on Login Page', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <BrowserRouter>
+          <LoginPage />
+        </BrowserRouter>
+      </Provider>
+    );
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+  it('it should render Home Page without crashing', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <BrowserRouter>
+          <HomePage />
+        </BrowserRouter>
+      </Provider>
+    );
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('it should render 404 PAGE without crashing', () => {
+    const wrapper = mount(
+      <BrowserRouter>
+        <NotFoundPage />
+      </BrowserRouter>
+    );
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('should render without crashing', () => {
+    const wrapper = mount(
+      <BrowserRouter>
+        <Menu />
+      </BrowserRouter>
+    );
+
+    expect(toJson(wrapper)).toMatchSnapshot();
+    expect(wrapper.find('[to="/"]')).toHaveLength(4);
+    expect(wrapper.find('Link').length).toBeGreaterThan(1);
+    expect(wrapper.find('ul')).toHaveLength(1);
+    expect(wrapper.find('li').length).toBeGreaterThan(1);
```

Comment:
could there be an assertion as to how many "li' you are expecting to be rendered?